src/main/java/core/basesyntax/service/impl/TransactionParserImpl.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.TransactionParser;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TransactionParserImpl implements TransactionParser {
+    @Override
+    public List<FruitTransaction> parse(List<String> transactionStrings) {
+        if (transactionStrings == null || transactionStrings.isEmpty()) {
+            throw new RuntimeException("Transactions parsing failed: "
+                    + "no transactions found");
```

Comment:
The exception message 'no transactions found' could be more informative by including suggestions for the user, such as checking if the file is empty or if the correct file is being parsed.